supreme_court_predictions/models/logistic_regression.py

```diff
@@ -1,48 +1,101 @@
 """
-TODO: Need file document string
+This LogisticRegression class runs logistic regression 
+on utterance data from the Supreme Court dataset. This class aims to predict
+the results of a case based on the text learned from utterances. 
 """
 
 import numpy as np
 import pandas as pd
 from sklearn.feature_extraction.text import CountVectorizer
 from sklearn.linear_model import LogisticRegression as skLR
+from sklearn.metrics import accuracy_score
 from sklearn.model_selection import train_test_split
 
 from supreme_court_predictions.util.files import get_full_data_pathway
 
 
 class LogisticRegression:
     """
-    TODO: Need document string
+    A class that runs logistic regression on aggregated utterance and cases data
+    from the Supreme Court dataset.
     """
 
     def __init__(self):
-        self.local_path = get_full_data_pathway("clean_convokit/")
+        self.local_path = get_full_data_pathway("processed/")
 
-        self.utterances_df = pd.read_pickle(self.local_path + "utterances_df.p")
-        self.logistic_regression()
+        self.total_utterances = pd.read_pickle(
+            self.local_path + "case_aggregations.p"
+        )
+        self.advocate_utterances = pd.read_pickle(
+            self.local_path + "advocate_aggregations.p"
+        )
+        self.adversary_utterances = pd.read_pickle(
+            self.local_path + "adversary_aggregations.p"
+        )
+        self.judge_utterances = pd.read_pickle(
+            self.local_path + "judge_aggregations.p"
+        )
+
+        self.run_regression()
 
-    def logistic_regression(self):
+    def logistic_regression(self, df):
         """
-        TODO: Need document string
+        Perform logistic regression on the given dataframe of utterance data.
+        It regresses on the entire dataset and regresses for judges, advocates,
+        and adversaries.
+
+        :param
+            df (pd.DataFrame): DataFrame containing utterance data
+
+        :return
+            float: Accuracy score of the logistic regression model
         """
-        vectorizer = CountVectorizer()
-        vectorize_document = self.utterances_df.loc[:, "tokens"].apply(" ".join)
+        vectorizer = CountVectorizer(analyzer="word", max_features=5000)
+        vectorize_document = df.loc[:, "tokens"].apply(" ".join)
         print("Creating bag of words")
         bag_of_words_x = vectorizer.fit_transform(vectorize_document)
 
-        # TODO: after Chay's merge of dataframe, this can be updated.
-        bag_of_words_y = np.random.randint(0, 2, len(vectorize_document))
+        bag_of_words_y = df.loc[:, "win_side"]
 
         X_train, X_test, y_train, y_test = train_test_split(
-            bag_of_words_x, bag_of_words_y, test_size=0.25, random_state=123
+            bag_of_words_x,
+            bag_of_words_y,
+            test_size=0.20,
+            random_state=123,
+            stratify=bag_of_words_y,
```

Comment:
try catch block and stratify allow smaller datsets to be used. Now the actual LR is done.